test/cnudie/cnudie_util_test.py

```diff
@@ -75,23 +75,65 @@ def test_componentdiff_to_str(cid):
         names_only_right={'c'},
         names_version_changed={'d'},
     )
-    result = cnudie.util.format_component_diff(
+    formatted_diff = cnudie.util.format_component_diff(
         component_diff=diff,
         delivery_dashboard_url_view_diff=None,
         delivery_dashboard_url=None,
     )
 
-    max_length = 10
-    if len(result) > max_length:
-        truncated_result = result[:result.find('## Component Details:')]
-        truncated_result += '\n... [Component details omitted]\n'
-        body = truncated_result
-    else:
-        body = result
+    release_notes = 'Release Notes\n' + 'A' * 70000
+
+    max_pr_body_length = 65536
+
+    max_body_length_exceeded_remark = (
+        '\n\nRelease notes were shortened since they exceeded the maximum length allowed for a '
+        'pull request body. The remaining release notes will be added as comments to this PR.'
+    )
+
+    combined_length = len(formatted_diff) + len(release_notes)
 
-    assert '### Added Components:' in body
-    assert '... [Component details omitted]' in body
-    assert '## Component Details:' not in body
+    if combined_length > max_pr_body_length:
+        # remove component details from formatted_diff
+        truncated_diff = formatted_diff[:formatted_diff.find('## Component Details:')]
```

Comment:
how about adding a switch to the function generating the diff and simply re-generate it (with truncation set to true on the second try?)

Reply:
also: why did you duplicate so much code into unittest?